script/main.js

```diff
@@ -0,0 +1,55 @@
+const bookList = document.querySelector('#bookList');
+const bookTitle = document.querySelector('#bookTitle');
+const bookAuthor = document.querySelector('#authorName');
+const addBook = document.querySelector('#addBook');
+const bookCollection = JSON.parse(localStorage.getItem('booklist')) ?? [];
+
+function displayBook() {
+  bookList.innerHTML = '';
+  bookCollection.forEach((book) => {
+    const p1 = document.createElement('p');
+    p1.textContent = `${book.title}`;
+    const p2 = document.createElement('p');
+    p2.textContent = `${book.author}`;
+    const indexOfBook = document.createElement('p');
+    indexOfBook.id = `bookindex${book.id}`;
+    indexOfBook.textContent = book.id;
+    indexOfBook.classList.add('hidden');
+    const removeButton = document.createElement('button');
+    removeButton.textContent = 'Remove';
+    removeButton.classList.add('deleteBtn');
+    const hr = document.createElement('hr');
+    bookList.append(p1, p2, indexOfBook, removeButton, hr);
+  });
+
+  const removeBtn = document.querySelectorAll('.deleteBtn');
+  removeBtn.forEach((elem) => {
+    elem.addEventListener('click', function removeBook() {
+      const getId = this.previousSibling.textContent;
+      const indexToRemove = bookCollection.findIndex((book) => book.id === Number(getId));
+
+      if (indexToRemove !== -1) {
+        bookCollection.splice(indexToRemove, 1);
```

Comment:
Well done on your implementation! It's evident that you've successfully incorporated the removal of books using their respective IDs, which is an important implimentation.

- [ ] [Optional] However, I'd like to suggest a small change that could enhance your code quality. Currently, you are using the `array.splice()` method to remove elements from the array. While this approach works as intended, I recommend considering the use of the `array.slice()` method instead. The [slice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) method of [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) instances returns a [shallow copy](https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy) of a portion of an array, selected from the start index to the end index (end index not included). This new array copy includes the items within the specified range, leaving the original array untouched. The key advantage of using `array.slice()` for this purpose is that it maintains the immutability of the original array. This means that the removal of a book using `array.slice()` won't modify the existing array but will return a new array with the desired items removed. This aligns well with the principles of data immutability and can lead to more predictable and maintainable code.

In summary, while both `array.splice()` and `array.slice()` are valid methods for removing elements, the choice to use `array.slice()` can contribute to clearer code that adheres to the principles of immutability. Keep up the excellent work, and happy coding! 🌟📝🚀